File: university-service/src/test/java/ua/com/foxminded/courseproject/validation/WeekScheduleDtoTest.java
package ua.com.foxminded.courseproject.validation;

import org.junit.jupiter.api.Test;
import ua.com.foxminded.courseproject.dto.WeekScheduleDto;

import javax.validation.ConstraintViolation;
import java.util.Set;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class WeekScheduleDtoTest extends ValidationSetupTest {

    @Test
    public void numberValidation_shouldThrowException_whenNumberIsNull() {
        WeekScheduleDto weekScheduleDto = new WeekScheduleDto();
        weekScheduleDto.setIsOdd(null);
        String expectedMessage = "must not be null";
        Integer expectedSize = 1;

        Set<ConstraintViolation<WeekScheduleDto>> constraintViolations = validator.validate(weekScheduleDto);
        Integer actualSize = constraintViolations.size();
        String actualMessage = constraintViolations.iterator().next().getMessage();

        assertEquals(expectedMessage, actualMessage);
        assertEquals(expectedSize, actualSize);
    }


    @Test
    public void numberValidation_shouldNotThrowException_whenNumberEqualsZero() {
        WeekScheduleDto weekScheduleDto = new WeekScheduleDto();
        weekScheduleDto.setIsOdd(true);
        Integer expectedSize = 0;

        Set<ConstraintViolation<WeekScheduleDto>> constraintViolations = validator.validate(weekScheduleDto);
        Integer actualSize = constraintViolations.size();

        assertEquals(expectedSize, actualSize);
    }

}

File: university-service/src/test/kotlin/ua/com/foxminded/courseproject/validation/WeekScheduleDtoTest.kt
package ua.com.foxminded.courseproject.validation

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import ua.com.foxminded.courseproject.dto.WeekScheduleDto

class WeekScheduleDtoTest : ValidationSetupTest() {
    @Test
    fun numberValidation_shouldThrowException_whenNumberIsNull() {
        val weekScheduleDto = WeekScheduleDto()
        weekScheduleDto.isOdd = null
        val expectedMessage = "must not be null"
        val expectedSize = 1
        val constraintViolations = validator.validate(weekScheduleDto)
        val actualSize = constraintViolations.size
        val actualMessage = constraintViolations.iterator().next().message
        Assertions.assertEquals(expectedMessage, actualMessage)
        Assertions.assertEquals(expectedSize, actualSize)
    }

    @Test
    fun numberValidation_shouldNotThrowException_whenNumberEqualsZero() {
        val weekScheduleDto = WeekScheduleDto()
        weekScheduleDto.isOdd = true
        val expectedSize = 0
        val constraintViolations = validator.validate(weekScheduleDto)
        val actualSize = constraintViolations.size
        Assertions.assertEquals(expectedSize, actualSize)
    }
}
